### PR TITLE
Add documentation for using buttons as links

### PR DESCRIPTION
**Pull request by calthomson**
# What's in this PR? 

This PR adds a small section to the Evergreen button documentation page: https://evergreen.segment.com/components/button explaining best practices for using the Button component as a link.

Open to feedback on phrasing, or whether to go into more or less depth.

Screen Shot 2019-03-11 at 5 45 58 PM

### CODE REVIEW

File: docs/src/pages/components/button.mdx [hunk 1]
```diff
@@ -398,6 +398,14 @@ The `BackButton` component is a preset based on the `Button` component.
 
 <PropsTable of="BackButton" />
 
+# Buttons as links
+
+Buttons can be used as links by using the optional `is` property with a value such as `ReactRouterLink` from [react-router](https://github.com/ReactTraining/react-router), or `a`.
```
Comment (vadimdemedes) [inline docs/src/pages/components/button.mdx]: I think it's just `Link` from `react-router`, not `ReactRouterLink`.

Comment (vadimdemedes) [reply docs/src/pages/components/button.mdx]: Could you also add ", or `a` tag." at the end?

**Comment by vadimdemedes** [comment]
Great addition, attention to detail 👍

File: docs/src/pages/components/button.mdx (hunk 1 above)
Comment (calthomson) [reply docs/src/pages/components/button.mdx]: For sure, thanks!

Comment (calthomson) [reply docs/src/pages/components/button.mdx]: Added.

**Pull request merged**